atom/common/native_mate_converters/net_converter.cc

```diff
@@ -191,16 +191,29 @@ bool Converter<net::HttpResponseHeaders*>::FromV8(
     if (!keys->Get(i)->ToString(context).ToLocal(&key)) {
       return false;
     }
-    if (!headers->Get(key)->ToString(context).ToLocal(&value)) {
-      return false;
+
+    if (headers->Get(key)->IsArray()) {
+      auto values = v8::Local<v8::Array>::Cast(headers->Get(key));
+      for (uint32_t j = 0; j < values->Length(); j++) {
+        if (!values->Get(j)->ToString(context).ToLocal(&value)) {
+          return false;
+        }
+
+        std::string k, v;
+        mate::ConvertFromV8(isolate, key, &k);
+        mate::ConvertFromV8(isolate, value, &v);
+        out->AddHeader(k + ": " + v);
```

Comment:
I do agree with @MarshallOfSound that we can avoid code repetition.

Maybe a small lambda would work here, like this:

```cc
bool Converter<net::HttpResponseHeaders*>::FromV8(
    v8::Isolate* isolate,
    v8::Local<v8::Value> val,
    net::HttpResponseHeaders* out) {
  if (!val->IsObject()) {
    return false;
  }

  auto addHeaderFromValue = [&isolate, &out](
      const v8::Local<v8::String>& key,
      const v8::Local<v8::Value>& localVal) {
    auto context = isolate->GetCurrentContext();
    v8::Local<v8::String> localStrVal;
    if (!localVal->ToString(context).ToLocal(&localStrVal)) {
      return false;
    }
    std::string k, v;
    mate::ConvertFromV8(isolate, key, &k);
    mate::ConvertFromV8(isolate, localStrVal, &v);
    out->AddHeader(k + ": " + v);
    return true;
  };

  auto headers = v8::Local<v8::Object>::Cast(val);
  ...
```